src/linesearch.jl

```diff
@@ -354,11 +357,25 @@ function perform_line_search(
     niter = 0
     α = line_search.alpha
     clipping = false
-    while f(x_storage) - f(x) > -gamma * α * dot_dir + α^2 * gamma^2 * ndir2 * M / 2 + eps(float(gamma)) &&
-              gamma ≥ 100 * eps(float(gamma))
+    relaxed_smoothness = line_search.relaxed_smoothness
+
+    gradient_storage = similar(gradient)
+    grad!(gradient_storage, x_storage)
+    dott = -fast_dot(gradient_storage-gradient, d)
```

Comment:
```suggestion
    dott = fast_dot(gradient, d) - fast_dot(gradient_storage, d)
```
otherwise we are allocating one vector